### PR TITLE
 MarineResearch

### PR DESCRIPTION
**Pull request by haludi**

### CODE REVIEW

File: Scenarios/MarineResearch/MarineResearchTest.cs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+            var client = new HttpClient();
+            var url =
+                $"{DocumentStore.Urls[0]}/databases/{DocumentStore.Database}/streams/queries?format=csv&query={Uri.EscapeDataString(query)}";
+            Stream stream = null;
+            async Task Action() => stream = await client.GetStreamAsync(url);
+            await Retry(5, Action, "export daily result to csv");
+            return stream;
+        }
+
+        private async Task Retry(int nTimes, Func<Task> action, string description)
```
Comment (ayende) [inline Scenarios/MarineResearch/MarineResearchTest.cs]: why do we need this?

Comment (haludi) [reply Scenarios/MarineResearch/MarineResearchTest.cs]: I try to simulate a real situation when you keep trying to send a request before you abort the mission
in situations of load and lose connection. Is it make sense if I will do it depend on the exception type?

**Comment by aviviadi** [comment]
@haludi - please close this PR, it is already merged in `master`. However later on conversions here should be addressed

File: Scenarios/MarineResearch/MarineResearchTest.cs (hunk 1 above)
Comment (ayende) [reply Scenarios/MarineResearch/MarineResearchTest.cs]: @aviviadi I'm not sure that I like all these retries.
This is meant to simulate client code, and I'm not sure they'll write retries.
We want to actually get these as test failures, no?

**Pull request closed without merging**